### DISCUSSION TITLE
How to use artefact generated by build stage in deployment stage ?

### DISCUSSION THREAD

**Discussion by dhwanitgupta** [aws/aws-cdk#23675] [Q&A]
Hi 

I am trying to create simple pipeline which deploy code to AWS Lambda. For that I have created below stages and steps

- Source ( CDK Package, LambdaCode Package)
- Build ( CDK Package Build, LambdaCode Package Build) -> Output (cdk.out,  S3 object)
- Self Mutate
- Deploy (Use Cloudfromation, Use S3 Object output of LambdaCode Build to deploy Lambda function)

I am not able to figure out methods using which I can use artefact generated/uploaded to S3 in Build stage while Deploying the Stage.

Adding Code

 ```
const serviceBuild = new ShellStep('Prebuild', {
      input: CodePipelineSource.codeCommit(Repository.fromRepositoryName(this, 'ASMCoreService', 'ASMCoreService'),
          'master'),
      primaryOutputDirectory: './build',
      commands: ['./build.sh']
    });

    const synthBuild = new ShellStep('PipelineSynthesis', {
      input: CodePipelineSource.codeCommit(
          Repository.fromRepositoryName(this, 'ASMPipelineCDKRepo', 'ASMPipelineCDK'),
           'master'),
      commands: ['npm ci', 'npm run build', 'npx cdk synth']
    });

    synthBuild.addStepDependency(serviceBuild);

    const pipeline = new CodePipeline(this, 'ASMServicePipeline', {
      pipelineName: 'ASMServicePipeline',
      synth: synthBuild,
      publishAssetsInParallel: true
    });

    const applicationStage = new ApplicationStage(this, "Gamma", props);

    pipeline.addStage(applicationStage);
```

In above code `serviceBuild` step build the package and deploy runtime zip file in S3 bucket. I want to use that S3 url in Application Stage to create Lambda Function.

**Comment by khushail** [comment]
Hi @dhwanitgupta , appreciate your patience on this ask. Could you please elaborate your ask whether you need artefact to be imported by next stage or you are looking for something else. 

If its the artefacts, CDK provides [codePipelinesAction](https://docs.aws.amazon.com/cdk/api/v1/docs/aws-codepipeline-actions-readme.html) construct where you can build output artefact and use those at next stage.Let me know if this fulfills your ask.
